drivers/sensor/lis2dh/lis2dh.c

```diff
@@ -421,6 +421,12 @@ int lis2dh_init(const struct device *dev)
 	LOG_INF("fs=%d, odr=0x%x lp_en=0x%x scale=%d", 1 << (LIS2DH_FS_IDX + 1), LIS2DH_ODR_IDX,
 		(uint8_t)LIS2DH_LP_EN_BIT, lis2dh->scale);
 
+	status = lis2dh->hw_tf->write_reg(dev, LIS2DH_REG_CTRL2, 0x01);
+	if (status < 0) {
+		LOG_ERR("Failed to set high pass filter");
+		return status;
+	}
+
```

Comment:
Recommend making this configurable with a devicetree property

Reply:
What you're doing here is configuring the hp filter for INT1 but 

- it's not clear without having read the datasheet. You should use and/or introduce appropriate #defines in the header and use those.
- you're configuring interrupt behavior, so you should do it in `lis2dh_trigger.c` instead
- you should make it configurable. A sensor-specific sensor attribute would be best, but a dts property would be ok too

Reply:
By the way that register is also used to control the cut-off frequency for the HP filter. See  https://www.st.com/resource/en/application_note/an5005-lis2dh12-mems-digital-output-motion-sensor-ultralowpower-highperformance-3axis-nano-accelerometer-stmicroelectronics.pdf